### DISCUSSION TITLE
cgroupsv2, containerd runtime and and docker image resource

### DISCUSSION THREAD

**Discussion by meezaan** [concourse/concourse#8131] [Help & Support] [answered]
Hello folks,

We run the concourse helm chart. Things were working great for over a year until Linode upgraded all their worker nodes with cgroupsv2 (or Debian Bullseye just came with it), and as a result everything broke. Switching the runtime to containerd helps solve some problems, but https://github.com/concourse/concourse-docker/issues/76 was getting in the way. I built my own Docker image with the fix in that PR and things got better.

But we use the docker image resource (https://github.com/concourse/docker-image-resource) to build Docker images, which does not work anymore because the runtime is using containerd. We get the followng error:

```
selected worker: concourse-worker-c474b9789-wswrs
waiting for docker to come up...
Error response from daemon: Get "https://our.concourse.domain/v2/": unauthorized: authentication required
```

Is there a way build Docker images with Concourse when the runtime is containerd?

Thank you!

**Comment by pn-santos** [comment, marked as answer]
> Is there a way build Docker images with Concourse when the runtime is containerd?

Yes: https://github.com/concourse/oci-build-task

**Reply by meezaan** [reply]
> > Is there a way build Docker images with Concourse when the runtime is containerd?
> 
> Yes: https://github.com/concourse/oci-build-task

Thank you. For now it fails with:

```
INFO[0000] building image                               
#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 2.46kB done
#1 DONE 0.0s
error: failed to solve: failed to read dockerfile: failed to mount /tmp/buildkit-mount120123706: [{Type:bind Source:/scratch/buildkitd/runc-native/snapshots/snapshots/1 Options:[rbind ro]}]: operation not permitted
FATA[0000] failed to build: build: exit status 1        
FATA[0000] failed to run task: exit status 1 
```

But I will investigate further.

**Reply by pn-santos** [reply]
Are you running it with `privileged: true` ?

**Reply by meezaan** [reply]
Doesn't look like it (doh!). I'll try with that.

**Reply by meezaan** [reply]
That did the trick. It works now. Thank you.
